.github/workflows/core_build_publish.yml

```diff
@@ -152,44 +151,40 @@ jobs:
           CORELIB_TESTS_AZURE_BLOB_STORAGE_CONTAINER_NAME: ${{ secrets.CORELIB_TESTS_AZURE_BLOB_STORAGE_CONTAINER_NAME }}
           CORELIB_TESTS_AZURE_TABLE_STORAGE_TABLE_NAME: ${{ secrets.CORELIB_TESTS_AZURE_TABLE_STORAGE_TABLE_NAME }}
       - name: Integration Test - SQL Server
-        run: dotnet test -c Release --logger trx --results-directory ../TestResults
-        working-directory: test/LeanCode.IntegrationTests
+        run: dotnet test -- --explicit on --filter-query '/[(category=integration)&(database!=postgres)]'
         env:
-          ASPNETCORE_TEST_CONTENTROOT_LEANCODE_INTEGRATIONTESTS: /home/runner/work/corelibrary/corelibrary/test/LeanCode.IntegrationTests
           SqlServer__ConnectionStringBase: Server=localhost,1433;User Id=sa;Password=Passw12#;Encrypt=false
           LeanCodeIntegrationTests__Database: sqlserver
       - name: Integration Test - Postgres
-        run: dotnet test -c Release --logger trx --results-directory ../TestResults
-        working-directory: test/LeanCode.IntegrationTests
+        run: dotnet test -- --explicit on --filter-trait 'category=integration'
         env:
-          ASPNETCORE_TEST_CONTENTROOT_LEANCODE_INTEGRATIONTESTS: /home/runner/work/corelibrary/corelibrary/test/LeanCode.IntegrationTests
           Postgres__ConnectionStringBase: Host=localhost;Username=postgres;Password=Passw12#
           LeanCodeIntegrationTests__Database: postgres
-      - name: Upload coverage to Codecov
-        uses: codecov/codecov-action@v3
-        if: always()
 
-      - name: Convert Test Results to jUnit
-        run: find -name '*.trx' -exec trx2junit --output TestResults {} +
-        working-directory: test
+      - name: Process test results
         if: always()
```

Comment:
Isn't it the same as without `if:`?

Reply:
Nope, the default for `if` is "if previous step succeeded", not `always`.